tools/handle_ocp_versions.py

```diff
@@ -1,18 +1,53 @@
 #!/usr/bin/env python3
 
+from argparse import ArgumentParser
 import json
 from pathlib import Path
+from typing import List
 
 from utils import check_output
 
-OCP_VERSIONS_FILE = Path("default_ocp_versions.json")
+OCP_VERSIONS_FILE = "default_ocp_versions.json"
+SKIP_IMAGES = ['registry.svc.ci.openshift.org/sno-dev/openshift-bip:0.2.0']
+
+
+def handle_arguments():
+    parser = ArgumentParser()
+    parser.add_argument('--src', type=str, default=OCP_VERSIONS_FILE)
+    parser.add_argument('--dest', type=str)
+    parser.add_argument('--ocp-override', type=str)
+
+    return parser.parse_args()
 
 
 def main():
-    with OCP_VERSIONS_FILE.open("r") as file_stream:
+    args = handle_arguments()
+
+    with Path(args.src).open("r") as file_stream:
         ocp_versions = json.load(file_stream)
 
+    if args.ocp_override:
+        update_openshift_versions_hashmap(ocp_versions, args.ocp_override)
+
     verify_ocp_versions(ocp_versions)
+    print(ocp_versions)
+
+    if args.dest:
```

Comment:
`else` dump to stdout instead

or make `args.dest` required

Reply:
It is always being printed before just for verbose. Do you think it should be only in case no `dest` was provided?

Reply:
Dump, not print. If the user doesn't provide and output file, it's reasonable to just write the result to stdout (and the above print should go to stderr).

Or alternatively just make the output file a required parameter

It's just that the way it is now, it's a rather "useless" flow to call this without an output parameter, so either make it useful by dumping to stdout or make it an error (param required)

Reply:
It is still useful to verify the source file validity. That was the original purpose of this script.
Can you explain the difference between `print` and `dump`?

Something like this?
```python
    if args.dest:
        ...
    else:
        print(json.dumps(...))
```

Reply:
Yes, like that - but make sure that *other* prints go to `stderr` if you do that.

If you dump I can just do

`./handle_ocp_versions.py --ocp-override ... --src default_ocp_versions.json`

to view the result or

`./handle_ocp_versions.py --ocp-override ... --src default_ocp_versions.json > my.json`

to write it to a file without having to bother with your parameters or

`./handle_ocp_versions.py --ocp-override ... --src default_ocp_versions.json | jq`

to pretty print it.

In general, when your script supports writing the result to stdout - it gives the user flexibility on what they want to do with the result, you don't force the result to go directly into a file.